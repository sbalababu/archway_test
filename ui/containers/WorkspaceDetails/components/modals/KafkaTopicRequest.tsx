import * as React from 'react';
import { InjectedFormProps } from 'redux-form';
import { Field } from 'redux-form/immutable';
import { reduxForm } from 'redux-form/immutable';
import FieldLabel from '../../../../components/FieldLabel';

/* tslint:disable:no-var-requires */
const { TextField } = require('redux-form-antd');

interface KafkaForm {
  name: string;
}

const KafkaTopicRequest = ({ handleSubmit }: InjectedFormProps<KafkaForm, {}>) => (
  <form style={{}} onSubmit={handleSubmit}>
    <FieldLabel>Topic Name</FieldLabel>
    <Field name="name" component={TextField} style={{ marginBottom: 0 }} />
  </form>
);

export default reduxForm<KafkaForm, {}>({
  form: 'topicRequest',
})(KafkaTopicRequest);
