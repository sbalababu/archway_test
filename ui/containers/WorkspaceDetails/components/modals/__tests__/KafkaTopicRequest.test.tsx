import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import KafkaTopicRequest from '../KafkaTopicRequest';

describe('KafkaTopicRequest', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<KafkaTopicRequest />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
