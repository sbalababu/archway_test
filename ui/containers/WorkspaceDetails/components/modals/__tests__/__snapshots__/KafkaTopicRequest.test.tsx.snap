// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KafkaTopicRequest renders correctly 1`] = `
<Hoc
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="topicRequest"
  forwardedRef={null}
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
/>
`;
